src/hotspot/os/bsd/os_perf_bsd.cpp

```diff
@@ -184,6 +217,65 @@ int CPUPerformanceInterface::CPUPerformance::cpu_loads_process(double* pjvmUserL
   _jvm_user_nanos = jvm_user_nanos;
   _jvm_system_nanos = jvm_system_nanos;
 
+  return result;
+#elif defined(__FreeBSD__)
+  struct kinfo_proc *lproc;
+  size_t length;
+  int mib[3] = { CTL_KERN, KERN_PROC, KERN_PROC_ALL };
+  int result = cpu_load_total_process(psystemTotalLoad);
+
+  if (sysctl(mib, 3, NULL, &length, NULL, 0) == -1) {
+    return OS_ERR;
+  }
+
+  lproc = (struct kinfo_proc *)malloc(length);
+  if (!lproc) {
+    return OS_ERR;
+  }
+
+  if (sysctl(mib, 3, lproc, &length, NULL, 0) == -1) {
+    free(lproc);
+    return OS_ERR;
+  }
+
+  int active_processor_count = os::active_processor_count();
+  long jvm_user_nanos = 0;
+  long jvm_system_nanos = 0;
+
+  long total_cpu_nanos;
+  if(!now_in_nanos(&total_cpu_nanos)) {
+    free(lproc);
+    return OS_ERR;
+  }
+
+  if (_total_cpu_nanos == 0 || active_processor_count != _active_processor_count) {
+    // First call or change in active processor count
+    result = OS_ERR;
+  }
+
+  for (size_t i = 0; i < length / sizeof(*lproc); i ++) {
+     struct rusage procusg = lproc[i].ki_rusage;
+     jvm_user_nanos += procusg.ru_utime.tv_usec + (1000000 * procusg.ru_utime.tv_sec);
+     jvm_system_nanos += procusg.ru_stime.tv_usec + (1000000 * procusg.ru_stime.tv_sec);
+  }
```

Comment:
FWIW, you could likely free lproc on the next line since it is unused for the rest of the function and that would save you having to do so in conditional blocks as well as at the end of the function.  In general I'd be in favour of extracting this information from lproc immediately after the sysctl call that populates it and freeing it then.  That would make it easier to reason about whether the memory handling is correct rather than needing to check for it in each inner block which features a 'return' (IMO).